root/dataframe/test_snapshot_copyaddresses.C

```diff
@@ -0,0 +1,63 @@
+// Regression test for issue https://github.com/root-project/root/issues/8295
+// Make sure that Snapshotting goes well when reading/writing data members of types saved as TBranchElements from
+// multiple input files (in particular, due to #8295, from the second file onwards Snapshot used to write wrong data).
+
+#include <ROOT/RDataFrame.hxx>
+#include <TError.h>
+#include <TFile.h>
+#include <TSystem.h>
+#include <TTree.h>
+
+constexpr static const char *fnamePrefix = "test_snapshot_copyaddresses";
+constexpr static int nfiles = 3;
+
+struct TwoInts {
+   int x;
+   int y;
+};
+
+void write_inputs()
+{
+   TwoInts ti{1, 2};
+   for (int i = 0; i < nfiles; ++i, ti.x += 2, ti.y += 2) {
+      const auto fname = fnamePrefix + std::to_string(i) + ".root";
+      if (gSystem->AccessPathName(fname.c_str()) == 0) // 0 means file already exists
+         continue;
+      TFile f(fname.c_str(), "recreate");
+      TTree t("t", "t");
+      t.Branch("ti", &ti);
+      t.Fill();
+      t.Write();
+   }
+}
+
+void test_snapshot_copyaddresses()
+{
+   write_inputs();
+
+   const std::string outSuffix = "_out.root";
+
+   ROOT::RDataFrame df("t", fnamePrefix + std::string("*.root"));
+   auto out_df = df.Snapshot<int, int>("t", fnamePrefix + outSuffix, {"x", "y"});
+
+   int expected = 1;
+   out_df->Foreach(
```

Comment:
Shouldn't you be (also) testing the content of the `Snapshot`ed output file?

Reply:
this is testing the contents of the snapshotted output file